lib/coach/middleware.rb

```diff
@@ -76,6 +76,15 @@ def instrument
       end
     end
 
+    def metadata
+      @_context[:metadata] ||= {}
```

Comment:
Is there anything we can do to prevent people from trying to use the metadata context themselves? It's the sort of key that might get used. (Or at least warn people somewhere.)